src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator, Drawable {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private String randomColor = colorSupplier.getRandomColor();
```

Comment:
You shouldn't have randomColor declared here. Let's pass random color to each figure constructor, as described [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#you-can-pass-random-values-to-the-constructor-of-a-figure-instead-of-generating-them-inside-figure-classes)